### PR TITLE
Add docstring example to VecOrMat

### PR DESCRIPTION
**Pull request by logankilpatrick**

### CODE REVIEW

File: base/array.jl [hunk 1]
```diff
@@ -64,10 +64,17 @@ Two-dimensional dense array with elements of type `T`, often used to represent
...
+
 """
     VecOrMat{T}
 
 Union type of [`Vector{T}`](@ref) and [`Matrix{T}`](@ref).
+
+# Examples
+```jldoctest
+julia> VecOrMat{Float64}
+Union{Array{Float64,1}, Array{Float64,2}}
```
Comment (kimikage) [inline base/array.jl:76]:
```suggestion
julia> Vector{Float64} <: VecOrMat{Float64}
true

julia> Matrix{Float64} <: VecOrMat{Float64}
true

julia> Array{Float64,3} <: VecOrMat{Float64}
false
```
Julia v1.6 displays aliases, so the original example is a kind of tautology. The tautology itself isn't that bad, but as you've given an example, the doctest is fragile.

Comment (logankilpatrick) [reply base/array.jl]: Great, adding in the example you suggested.

**Comment by ViralBShah** [comment]
I am a bit borderline on this one. I am not sure about the examples here - if they add much to the explanation. They are not an example of the actual code, but a demonstration. Wondering what others think?

**Comment by ViralBShah** [comment]
Revisiting this, I think this is worth merging. But it would be great to have a couple of lines that say what this type is for - a union type that allows functions to accept either a matrix or a vector. It was originally for internal use only - and documentation means we are committing to it, in a way.

**Comment by logankilpatrick** [comment]
@ViralBShah Okay, added.

**Comment by logankilpatrick** [comment]
(Note I originally just opened this since it showed up on the Julia Docs but didn't have a doctoring)

**Comment by vtjnash** [comment]
I was also skeptical of this too, but seeing that Viral then approved, I've merged this.

**Pull request merged**